romanisim/tests/test_l3.py

```diff
@@ -142,3 +143,364 @@ def test_inject_sources_into_mosaic():
+            None, galsim.Sersic(1, half_light_radius=0.2) * vega_sed, None)
+    ]
+    tabcat = table.Table()
+    tabcat['ra'] = [270.0]
+    tabcat['dec'] = [66.0]
+    tabcat[filter_name] = counts
+    tabcat['type'] = 'PSF'
+    tabcat['n'] = -1
+    tabcat['half_light_radius'] = -1
+    tabcat['pa'] = -1
+    tabcat['ba'] = -1
+
+    # Return dictionary with the above values
+    return dict(im=im, impsfgray=impsfgray,
+                impsfchromatic=impsfchromatic,
+                bandpass=bandpass, counts=counts, fluxdict=fluxdict,
+                graycatalog=graycatalog,
+                chromcatalog=chromcatalog, filter_name=filter_name,
+                tabcatalog=tabcat)
+
+
+def test_simulate_vs_cps():
+    """ Tests to ensure that simulate runs match simulate_cps output
+    """
+    # Set random seed
+    rng_seed = 42
+
+    # Set image, catalog, etc.
+    imdict = set_up_image_rendering_things()
+    chromcat = imdict['chromcatalog']
+    graycat = imdict['graycatalog']
+    coord = SkyCoord(270 * u.deg, 66 * u.deg)
+    for o in chromcat:
+        o.sky_pos = coord
+    for o in graycat:
+        o.sky_pos = coord
+    # these are all dumb coordinates; the coord sent to simulate_counts
+    # is the coordinate of the boresight, but that doesn't need to be on SCA 1.
+    # But at least they'll exercise some machinery if the ignore_distant_sources
+    # argument is high enough!
+    roman.n_pix = 100
+    exptime = 600
+
+    # Create metadata
+    meta = util.default_image_meta(filter_name='F158')
+    wcs.fill_in_parameters(meta, coord)
+    metadata = deepcopy(parameters.default_mosaic_parameters_dictionary)
+    filter_name = 'F158'
+    metadata['basic']['optical_element'] = filter_name
+    metadata['wcsinfo']['ra_ref'] = 270
+    metadata['wcsinfo']['dec_ref'] = 66
+
+    # Set up blank image
+    im = imdict['im'].copy()
+    im.array[:] = 0
+
+    # Set WCS
+    twcs = romanisim.wcs.get_mosaic_wcs(metadata, shape=(roman.n_pix, roman.n_pix))
+
+    # Create chromatic data in simulate_cps
+    im1 = im.copy()
+    im1, extras1 = l3.simulate_cps(im1, metadata, exptime, objlist=chromcat,
+                                   xpos=[50] * len(chromcat), ypos=[50] * len(chromcat),
+                                   bandpass=imdict['bandpass'], seed=rng_seed,
+                                   ignore_distant_sources=100)
+
+    # Create filter data in simulate_cps
+    im2 = im.copy()
+    im2, extras2 = l3.simulate_cps(im2, metadata, exptime, objlist=graycat,
+                                   xpos=[50] * len(chromcat), ypos=[50] * len(chromcat),
+                                   psf=imdict['impsfgray'], seed=rng_seed,
+                                   ignore_distant_sources=100)
+
+    # Ensure that the two simualtions are in line with each other
+    maxim = np.where(im1.array**2 > im2.array**2, im1.array, im2.array)
+    m = np.abs(im1.array**2 - im2.array**2) <= 20**2 * np.abs(maxim)
+    assert np.all(m)
```

Comment:
This looks like a huge, huge range?  You probably want abs(im1 - im2) < something, but 400 times the maximum of the image is weird?  But it's the difference of squares, so the units are different?  I don't think we actually need to compare chromatic vs. filter in the first place.

Reply:
Removed this as the sim vs sim_cps test is the important one.